broadcaster/backlog/backlog.go

```diff
@@ -110,33 +107,30 @@ func (b *backlog) Get(start, end uint64) (*m.BroadcastMessage, error) {
 		return nil, errOutOfBounds
 	}
 
-	if start < head.Start() {
-		start = head.Start()
-	}
-
 	if end > tail.End() {
 		return nil, errOutOfBounds
 	}
 
-	found, err := b.Lookup(start)
-	if err != nil {
+	segment, err := b.Lookup(start)
+	if start < head.Start() {
+		// doing this check after the Lookup call ensures there is no race
+		// condition with a delete call
```

Comment:
I'm not sure this quite works because `head` was already loaded before in `b.head.Start()`. Maybe you could reload `head` after the `Lookup` call?

Reply:
Good shout, changed